### ISSUE TITLE
CatalogImportExport multiselect values will fail validation with custom multiValueSeperator

### ISSUE THREAD

**Issue by koenner01** [magento/magento2#6984]
If you are trying to import product data with the CatalogImportExport interface, validation on multiselect attributes will fail if you are using a multi value separator different from '|'.
### Preconditions
1. Magento 2.1
2. PHP7.0
### Steps to reproduce
1. Create a catalog_product csv that contains product data and at least 1 multiselect attribute with 2 options (and separate them with custom separator)
2. Go to the MG backend Import interface
3. Set your custom delimiter as 'Multiple value separator' (example separator '|||')
4. Choose your file and click the 'Check Data' button 
### Expected result
1. Validation should pass without problems
### Actual result
1. Validation fails and tells us we have incorrect values for our multiselect attribute

I have traced this problem back to \Magento\CatalogImportExport\Model\Import\Product\Validator. In the function 'isAttributeValid' the following code is present:

```
case 'multiselect':
    $values = explode(Product::PSEUDO_MULTI_LINE_SEPARATOR, $rowData[$attrCode]);
    $valid = true;
    foreach ($values as $value) {
        $valid = $valid && isset($attrParams['options'][strtolower($value)]);
    }
    if (!$valid) {
        $this->_addMessages(
            [
                sprintf(
                    $this->context->retrieveMessageTemplate(
                        RowValidatorInterface::ERROR_INVALID_ATTRIBUTE_OPTION
                    ),
                    $attrCode
                )
            ]
        );
    }
    break;
```

As you can see the values are being seperated by a hardcoded constant defined in \Magento\CatalogImportExport\Model\Import\Product which is '|'. So if your separator for multiselect values is different from the hardcoded value, validation will fail (**even though you set the correct value in 'Multiple value separator'**)

The hardcoded reference should be changed to something like

```
$parameters = $this->context->getParameters();
$separator = (isset($parameters[\Magento\ImportExport\Model\Import::FIELD_FIELD_MULTIPLE_VALUE_SEPARATOR])
    ? $parameters[\Magento\ImportExport\Model\Import::FIELD_FIELD_MULTIPLE_VALUE_SEPARATOR]
    : Product::PSEUDO_MULTI_LINE_SEPARATOR)
;
$values = explode($separator, $rowData[$attrCode]);
```

**Comment by koenner01** [comment]
Little addendum to this; There are 4 files where this hardcoded separator value is being (mis)used

\Magento\CatalogImportExport\Model\Export\Product
\Magento\CatalogImportExport\Model\Import\Product\Option
\Magento\CatalogImportExport\Model\Import\Product\Type\AbstractType
\Magento\CatalogImportExport\Model\Import\Product\Validator

They should all be fixed.

**Comment by dthampy** [comment]
@This looks an issue that already exists. Internal jira ticket - MAGETWO-58977( Target version- 2.1.3)

**Comment by hostep** [comment]
@dthampy or @veloraven: can you confirm this is fixed in 2.1.3?
I see these two commits which seem to have something to do with this issue: https://github.com/magento/magento2/commit/1ab5048f9aaa6e7584c696d84d69bf50f0234bbb & https://github.com/magento/magento2/commit/c6c16165b4bbd84d5800f1419cbe61321bc33dd4, but maybe I'm wrong?

If this is fixed in 2.1.3, you can close this ticket :)

**Comment by agoeurysky** [comment]
@koenner01 @dthampy @veloraven @hostep There is the same issue with `\Magento\GroupedImportExport\Model\Import\Product\Type\Grouped` :

`$associatedSkusAndQtyPairs = explode(Import::DEFAULT_GLOBAL_MULTI_VALUE_SEPARATOR, $associatedSkusQty);`

**Comment by pboisvert** [comment]
https://github.com/magento/magento2ce/pull/544 is the pull request for this issue. Issues shows as closed fixed against 2.1.3 so closing. 

@antboiko can you check issue mentioned in last comment for grouped products and open a new magetwo ticket if needed

**Comment by koenner01** [comment]
@pboisvert the pull request link your are referring to isn't working ?
